### PR TITLE
Norm Fasey - Chitter Challenge

### PR DESCRIPTION
**Pull request by DeathRay1977**

### CODE REVIEW

**Comment by tansaku** [comment]
You had your chitter_challenge reviewed by **jamie**.
### The good points are:
* Correct installation instructions in README
* All unnecessary components removed, Ruby style guide conventions followed
* Rakefile has appropriate tasks
* spec_helper.rb pulls in entire app through single point, Only test helpers (not view helpers) are pulled in to spec_helper.rb
* Database Cleaner is set up correctly
* DataMapper set up in separate configuration file
* Feature tests organized correctly
* Feature and unit tests in appropriate locations
* Spec 'Helpers' used appropriately
* Asset routes set correctly
* Used modular style
* flash.now and flash.next used correctly
* All controller methods short and succinct
* Routes split over separate controller files
* Forms use correct markup
* Correct use of Semantic HTML elements
* Partials used appropriately
* View Helpers used appropriately

### You should consider the following improvements:
* Gemfile should use Test Groups

see https://github.com/makersacademy/chitter_challenge/blob/master/docs/review.md for more details

File: app/controllers/peeps.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  get '/peeps/index' do
+    @username = current_user ? current_user.username : nil
+    messages = Peep.all(order: [:created_at.desc])
+    @peeps = []
+    messages.each do |message|
+      @peeps << {
+        text: message.text,
+        author: User.get(message.user_id).username,
+        time: message.created_at.strftime("%a, %e %b %Y %H:%M:%S %z")
+    }
```
Comment (danldb) [inline app/controllers/peeps.rb]: This could all be extracted to your models. Try to keep your controllers skinny.

Comment (DeathRay1977) [inline app/controllers/peeps.rb]: Done...

File: spec/features/sign_in_with_valid_credentials_spec.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+feature 'Signing in' do
+  scenario 'with valid credentials' do
+    sign_in_with_valid_credentials
+    expect(page).to have_content 'Welcome, JohnDoe!'
```
Comment (danldb) [inline spec/features/sign_in_with_valid_credentials_spec.rb]: JohnDoe is a magic string - you could have your sign_in method accept arguments for name, password or whatever, in order to make it clear that you are expecting the users name to show on the page.

File: spec/user_spec.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
+require_relative '../app/models/user'
+
+describe User do
+  subject(:user) {described_class}
+  let(:email) {'example@example.com'}
+  let(:password) {'TopS3cr3t'}
+  before(:each) do
+    @valid_user = user.create(email: email,
```
Comment (danldb) [inline spec/user_spec.rb]: consider using let instead of an instance variable

**Comment by danldb** [comment]
:checkered_flag: 

Nice submission Norm, thanks :)

**Pull request closed without merging**